src/Data/Nat/Properties.agda

```diff
@@ -1017,11 +1020,26 @@ m^n≡1⇒n≡0∨m≡1 m (suc n) eq = inj₂ (m*n≡1⇒m≡1 m (m ^ n) eq)
 ⊔-idem : Idempotent _⊔_
 ⊔-idem = sel⇒idem ⊔-sel
 
+⊔-greatest : ∀ {m n o} → m ≥ o ⊎ n ≥ o → m ⊔ n ≥ o
+⊔-greatest {zero} {n} (inj₁ z≤n) = z≤n
```

Comment:
Feels like there should be a vastly shorter proof by using [`⊔-sel`](http://agda.github.io/agda-stdlib/Data.Nat.Properties.html#28082)

Reply:
In fact we've already proved the two constituent parts `m≤n⇒m≤n⊔o` and `m≤n⇒m≤o⊔n` so all you have to do is put them together.